taichi/codegen/cuda/codegen_cuda.cpp

```diff
@@ -267,6 +267,14 @@ class TaskCodeGenCUDA : public TaskCodeGenLLVM {
       auto frac = call("__nv_frexp", double_input, exp_ptr);
       builder->CreateStore(frac, frac_ptr);
       llvm_val[stmt] = res;
+    } else if (op == UnaryOpType::popcnt) {
+      if (input_taichi_type->is_primitive(PrimitiveTypeID::i32)) {
+        llvm_val[stmt] = call("__nv_popc", input);
+      } else if (input_taichi_type->is_primitive(PrimitiveTypeID::i64)) {
+        llvm_val[stmt] = call("__nv_popcll", input);
```

Comment:
nit: support u32 / u64 as well. The input may need to be casted.

Reply:
@bobcao3  Did you mean convert input primitive type to unsigned first? then do nv_popc?

Reply:
or cast at front-ir?

Reply:
u32 and i32 are processed subsequently with another intrinsics. I think this part looks good